app/src/main/java/org/oppia/app/parser/StringToRatioParser.kt

```diff
@@ -0,0 +1,80 @@
+package org.oppia.app.parser
+
+import android.content.Context
+import androidx.annotation.StringRes
+import org.oppia.app.R
+import org.oppia.app.model.RatioExpression
+
+/** This class contains method that helps to parse string to ratio. */
+class StringToRatioParser {
+  private val invalidCharsRegex =
+    """^[\d\s/\:\.]+$""".toRegex()
+  private val invalidRatioRegex =
+    """^(\s)*(\d+((\s)*:(\s)*\d+)+)(\s)*$""".toRegex()
+
+  /**
+   * Returns a [RatioParsingError] for the specified text input if it's an invalid ratio, or
+   * [RatioParsingError.VALID] if no issues are found. Note that a valid ratio returned by this method is guaranteed
+   * to be parsed correctly by [parseRatioFromString].
+   *
+   * This method should only be used when a user tries submitting an answer. Real-time error detection should be done
+   * using [getRealTimeAnswerError], instead.
+   */
+  fun getSubmitTimeError(text: String, numberOfTerms: Int): RatioParsingError {
+    val ratio = parseRatioOrThrow(text)
+    return if (ratio.ratioComponentCount < numberOfTerms) {
+      RatioParsingError.INVALID_FORMAT
```

Comment:
Error Message for this error 
INVALID_SIZE : Number of terms is less than the required number of terms?

Reply:
What did we use for the web error here?

Reply:
@aggarwalpulkit596 did you miss this?

Reply:
Oh i missed this one and on web we use the above error message.